stl/inc/xutility

```diff
@@ -4198,7 +4198,383 @@ _NODISCARD _CONSTEXPR17 move_iterator<_Iter> make_move_iterator(_Iter _It) { //
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+        if (_Left._Mystate == _State::_Holds_iter && _Right._Mystate == _State::_Holds_iter) {
+            return _Left._Myiter == _Right._Myiter;
+        } else if (_Left._Mystate == _State::_Holds_sentinel && _Right._Mystate == _State::_Holds_sentinel) {
+            return true;
+        } else if (_Left._Mystate == _State::_Holds_iter && _Right._Mystate == _State::_Holds_sentinel) {
+            return _Left._Myiter == _Right._Mysentinel;
+        } else {
+            return _Left._Mysentinel == _Right._Myiter;
+        }
+    }
+
+    // clang-format off
+    template <sized_sentinel_for<_Iter> _Iter2, sized_sentinel_for<_Iter> _Se2>
+        requires sized_sentinel_for<_Se, _Iter2>
+    _NODISCARD friend iter_difference_t<_Iter2> operator-(
+        const common_iterator& _Left, const common_iterator<_Iter2, _Se2>& _Right) {
+        // clang-format on
+#if _ITERATOR_DEBUG_LEVEL != 0
+        _STL_VERIFY(_Left._Mystate != _State::_Nonthing && _Right._Mystate != _State::_Nonthing,
+            "Cannot take difference of valueless common_iterators");
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+        if (_Left._Mystate == _State::_Holds_iter && _Right._Mystate == _State::_Holds_iter) {
+            return _Left._Myiter - _Right._Myiter;
+        } else if (_Left._Mystate == _State::_Holds_sentinel && _Right._Mystate == _State::_Holds_sentinel) {
+            return 0;
+        } else if (_Left._Mystate == _State::_Holds_iter && _Right._Mystate == _State::_Holds_sentinel) {
+            return _Left._Myiter - _Right._Mysentinel;
+        } else {
+            return _Left._Mysentinel - _Right._Myiter;
+        }
+    }
+
+    // clang-format off
+    _NODISCARD friend iter_rvalue_reference_t<_Iter> iter_move(const common_iterator& _Right) noexcept(
+        noexcept(_RANGES iter_move(_STD declval<const _Iter&>())))
+        requires input_iterator<_Iter>
+    {
+        // clang-format on
+#if _ITERATOR_DEBUG_LEVEL != 0
+        _STL_VERIFY(
+            _Right._Mystate == _State::_Holds_iter, "Can only iter_move from common_iterator if it holds an interator");
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+        return _RANGES iter_move(_Right._Myiter);
+    }
+
+    template <indirectly_swappable<_Iter> _Iter2, class _Se2>
+    friend void iter_swap(const common_iterator& _Left, const common_iterator<_Iter2, _Se2>& _Right) noexcept(
+        noexcept(_RANGES iter_swap(_STD declval<const _Iter&>(), _STD declval<const _Iter2&>()))) {
+#if _ITERATOR_DEBUG_LEVEL != 0
+        _STL_VERIFY(_Left._Mystate == _State::_Holds_iter && _Right._Mystate == _State::_Holds_iter,
+            "Can only iter_swap from common_iterators if both hold an interator");
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+        return _RANGES iter_swap(_Left._Myiter, _Right._Myiter);
+    }
+
+private:
+    enum class _State : unsigned char { _Nonthing, _Holds_iter, _Holds_sentinel };

```

Comment:
Should `_Nonthing` be `_Nothing`?

Reply:
Thanks, although I have to say that non-thing is really close in that context